### ISSUE TITLE
Bitwardin 2023.12.0 Auto-Fill not working correctly

### ISSUE THREAD

**Issue by DjKilla2** [bitwarden/clients#7202]
### Steps To Reproduce

1. Go to many websites and use Auto-Fill to fill forms

### Expected Result

Auto-Fill worked with Bitwardin version 2023.10.2

### Actual Result

https://www.ferrerorocher.com/us/en/xp/giveagoldengreeting/  (form doesn't fill)
https://www.jrcigars.com/sweepstakes/  (address, city, zip code doesn't fill)
https://apfco.com/Secured/S9133/  (after entering birthdate, the following form only fills first and last name and nothing else)
https://www.littlepotatoes.com/holiday-happiness/  (form doesn't fill)
https://fireballfantasy.com/  (form doesn't fill)
https://www.quadratec.com/jeep-vehicle-giveaway  (form doesn't fill)

### Operating System

Windows

### Operating System Version

Windows 10 22H2 (64-bit)

### Web Browser

Firefox

### Browser Version

Firefox 120.0.1 (64-bit)

### Build Version

Firefox 120.0.1 (64-bit)

### Issue Tracking Info

- [ ] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Krychaz** [comment]
Hello there, 

Bitwarden version 2023.12 features an updated Autofill algorithm (V2), which is incompatible with the Firefox Browser extension versions 2023.10 and older. 

In order to resolve the issues that you’re seeing, could you please update your Firefox Browser extension as described in this article: https://support.mozilla.org/en-US/kb/how-update-add-ons

**Comment by cagonzalezcs** [comment]
@DjKilla2 

Thank you for your bug report, I was able to confirm the issue as reproducible within the recent extension update. 

Because these form fields are fading in on scroll after we initially capture the form elements on the page, we are presenting a cached set of data that incorrectly identifies the form fields as hidden elements. The extension does not autofill elements that are not viewable, so something needs to change to ensure form elements which fade in are updated to a viewable state.

I'm discussing this right now with the Autofill Team on how we will resolve the issue, will try to post an update in this thread when a path forward to resolve this has been decided.

**Comment by cagonzalezcs** [comment]
Furthermore, I found that we run into a similar visibility problem with forms that load outside of the viewport initially. 

For the moment, we will be keeping the autofill v2 flag on and will work on a fix to resolve the issues present in this thread. If a more critical set of issues present themselves, we will make the consideration of reverting the v2 flag. 

I'll follow up on this ticket by linking the PR which addresses this problem.

**Comment by DjKilla2** [comment]
Thank you for taking the time to confirm this issue. I did a complete uninstall of the latest version and installed the last version (2023.10.2) and the same thing happened, where as, it was working before. I did a sync but that didn't help, so I'm guessing the v2 flag is retroactive and affects older previous versions of the extension. Hope this additional info helps.

**Comment by cagonzalezcs** [comment]
Yes, the autofill v2 flag is what would impede resolution of this issue by using an older version of the extension. As I mentioned above, we currently plan on keeping the v2 flag in place for the moment. A number of updates to how autofill functions are incorporated with the recent build, and rolling back to v1 is possible but we'd like to avoid it if we can.

That said, for the issue described here there is a workaround I've found. If you scroll down to the form element and have it in view, then refresh the page such that the form element loads in view, then autofill will work as expected. 

Along with that, we've got a [PR in place](https://github.com/bitwarden/clients/pull/7209) that should resolve this issue by checking for field visibility each time we call to collectPageDetails and return cached page detail results. 

Due to the complicated and slow nature of third-party extension store updates, this fix will not be able to be available until the next release. However, we have accounted for it in a manner that resolves the problems within this ticket. 

I'm closing this ticket for now, if other concerns exist, please feel free to comment here and we can consider re-opening the ticket if necessary.
